src/main/groovy/com/lesfurets/jenkins/unit/declarative/AllOfDeclaration.groovy

```diff
@@ -0,0 +1,49 @@
+package com.lesfurets.jenkins.unit.declarative
+
+import org.springframework.util.AntPathMatcher
+
+class AllOfDeclaration extends WhenDeclaration {
+
+    List<String> branches = []
+    List<Boolean> expressions = []
+
+    def branch(String name) {
+        this.branches.add(name)
+    }
+
+    def expression(Closure closure) {
+        this.expressions.add(closure)
+    }
+
+    def expressions(delegate) {
+        def exp_result;
+        for (def exp in this.expressions) {
+            exp_result = executeWith(delegate, exp)
+            if (!exp_result) {
+                return false
+            }
+        }
+        return true
+    }
+
+    Boolean execute(Object delegate) {
+        AntPathMatcher antPathMatcher = new AntPathMatcher();
+
+        if (this.branches.size() > 0) {
+            for(def branch in this.branches) {
+                if (!antPathMatcher.match(branch, delegate.env.BRANCH_NAME) ) {
+                    return false
+                }
+            }
+        }
+
+        if (this.expressions.size() > 0) {
+            boolean exp = expressions(delegate)
+            if (!exp) {
+                return false
+            }
+        }
+
+        return true
+    }
+}
```

Comment:
Nit: missing a newline at the end of the file

Reply:
copied from existing file.  If this is a requirement it should be fixed in a linter.